common/src/main/java/com/wynntils/models/beacons/BeaconModel.java

```diff
@@ -80,46 +82,77 @@ public void onEntityAdd(AddEntityEvent event) {
 
     @SubscribeEvent
     public void onEntityTeleport(TeleportEntityEvent event) {
-        Optional<VerifiedBeacon> verifiedBeaconOpt = verifiedBeacons.stream()
-                .filter(verifiedBeacon -> verifiedBeacon.getBaseEntity().equals(event.getEntity()))
-                .findFirst();
+        if (event.getEntity() == null) return;
```

Comment:
I think we should use mixins so this never happens.

Reply:
Oh, sorry. This is an edge case, you can't do the invoke-shift trick I did before. Maybe only post the event if entity is not null?